### DISCUSSION TITLE
Is there a way to place items in a row with justify-content: space-between?

### DISCUSSION THREAD

**Discussion by martinjonsson01** [QuestPDF/QuestPDF#636] [Q&A] [answered]
In CSS you can use flexbox to have items in a row be allowed to grow (i.e. not fixed-size) while being positioned as far away from each other as possible with space inbetween.

You'd set the value:
```css
justify-content: space-between;
```
and get the following:
![image](https://github.com/QuestPDF/QuestPDF/assets/12572888/1ac93e25-2afa-4a82-8891-2a5b80f97934)

Is there a way to achieve this in QuestPDF? I've tried using a Table, but that's too rigid (I need the items to be dynamically sized, sometimes one is much wider and sometimes another is). I've also tried a normal Row, but that doesn't seem to allow the kind of layout I'm after. Using an Inlined component with `.AlignJustify()` almost worked, but it wants to wrap so it no longer contains the elements on a single row, which is not what I'm after.

**Comment by MarcinZiabek** [comment, marked as answer]
Please consider the following solution that alternates actual content elements with equally distributed relative spacing. If you plan to use this solution in more places, it should be pretty easy to create an extension method that abstracts the logic.

When using `AutoItem`, I recommend using at least the `MaxWidth` element to minimize the chance of layout-related exceptions. Of course, the safest option is the `Inlined` element, which prevents layout issues by wrapping its content to the new line.

```csharp
container
    .Padding(25)
    .MinimalBox()
    .ExtendHorizontal()
    .Border(1)
    .Row(row =>
    {
        row.AutoItem().Width(100).Height(50).Background(Colors.Red.Lighten2);
        row.RelativeItem();
        row.AutoItem().Width(150).Height(50).Background(Colors.Green.Lighten2);
        row.RelativeItem();
        row.AutoItem().Width(200).Height(50).Background(Colors.Blue.Lighten2);
    });
```

<img width="1864" alt="image" src="https://github.com/QuestPDF/QuestPDF/assets/9263853/55abcb0b-dad6-4163-8c00-ee575cdfd367">

**Reply by martinjonsson01** [reply]
That's an interesting suggestion, I'll try it out and let you know how it works!

**Reply by martinjonsson01** [reply]
That worked, thank you.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
